### PR TITLE
Add Search administration scripts and ListIndexes method

### PR DESCRIPTION
**Pull request by jeanleonov**
- Basic implementation of ListIndexes method.
 - List collections script was implemented.
 - Delete collection script was implemented.
 - Reindex collection (index) screip was implemented.

### CODE REVIEW

File: SearchService2/appscale/search/api_methods.py [hunk 1]
```diff
@@ -99,8 +99,15 @@ def __init__(self, zk_client):
       list_indexes_request: A search_pb2.ListIndexesRequest.
       list_indexes_response: A search_pb2.ListIndexesResponse.
     """
-    raise InvalidRequest("List indexes method is not implemented "
-                         "in AppScale SearchService2 yet")
+    app_id = list_indexes_request.app_id.decode('utf-8')
+    # TODO list indexes params should be proceeded
```
Comment (cdonati) [inline SearchService2/appscale/search/api_methods.py]: proceeded -> processed?

File: SearchService2/appscale/search/scripts.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      logger.info('Collections:\n    {}'.format('\n    '.join(collections)))
+      if broken:
+        logger.warning('Broken collections:\n    {}'.format('\n  '.join(broken)))
+    except (SolrServerError, SolrClientError) as err:
+      logger.error('Failed to list Solr collections ({}).'.format(err))
+    io_loop.stop()
+    io_loop.close()
+
+  io_loop = ioloop.IOLoop.current()
+  io_loop.spawn_callback(list_collections)
```
Comment (cdonati) [inline SearchService2/appscale/search/scripts.py]: Would `IOLoop.run_sync` be appropriate for this?

Comment (jeanleonov) [reply SearchService2/appscale/search/scripts.py]: Oh.. good old copy-paste=) Thanks, I'll update it

File: SearchService2/setup.py [hunk 3]
```diff
@@ -34,5 +34,8 @@
   include_package_data=True,
   entry_points={'console_scripts': [
     'appscale-search2=appscale.search.search_server:main',
+    'appscale-search2-reindex=appscale.search.scripts:reindex',
+    'list-solr-collections=appscale.search.scripts:list_solr_collections',
+    'delete-solr-collection=appscale.search.scripts:delete_solr_collection',
   ]}
```
Comment (sjones4) [inline SearchService2/setup.py]: Commands we own should probably start with `appscale-`

File: SearchService2/appscale/search/scripts.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  parser.add_argument(
+    '-v', '--verbose', action='store_true',
+    help='Output debug-level logging')
+  parser.add_argument(
+    '--no-prompt', action='store_true',
+    help='')
+  parser.add_argument(
+    '--zk-locations', required=True, nargs='+', help='ZooKeeper location(s)')
+  parser.add_argument(
+    '--collection', required=True, help='ZooKeeper location(s)')
```
Comment (sjones4) [inline SearchService2/appscale/search/scripts.py]: help text copy/paste

File: SearchService2/setup.py (hunk 3 above)
Comment (jeanleonov) [reply SearchService2/setup.py]: Thanks for the review!
I see that it breaks our usual commands style. But I think it is acceptable as the package should be installed in venv, so these binaries won't be in a global namespace.

Though I don't have a strong opinion on this.. would you still prefer to rename it?

Comment (sjones4) [reply SearchService2/setup.py]: I think it is preferable to use consistent  / discoverable names where possible. I also think it is fine to merge without changes, this is just nice to have.

**Comment by jeanleonov** [comment]
http://ocd.appscale.net:8080/job/Daily%20Build/6738/

**Comment by jeanleonov** [comment]
http://ocd.appscale.net:8080/job/Daily%20Build/6740/

**Pull request merged**